### PR TITLE
Stanford dependency parser

### PR DESCRIPTION
**Pull request by rmalouf**
Modify interface to the Stanford Parser to return either constituency trees or typed
dependency graphs.

### CODE REVIEW

File: nltk/parse/dependencygraph.py [hunk 1]
```diff
@@ -334,6 +334,7 @@ def extract_10_cells(cells):
             # Make sure that he fake root node has labeled dependencies.
             if (cell_number == 3) and (head == 0):
                 rel = 'ROOT'
+            if rel == 'root': rel = 'ROOT'  # Stanford parser doesn't capitalize ROOT
```
Comment (dimazest) [inline nltk/parse/dependencygraph.py]: Could you please split it to several lines (according to pap8 guidelines):

``` python
# Stanford parser doesn't capitalize ROOT
if rel == 'root':
    rel = 'ROOT'  
```

**Comment by stevenbird** [comment]
@rmalouf – nice to have the extra doctests, thanks.

File: nltk/parse/dependencygraph.py (hunk 1 above)
Comment (rmalouf) [inline nltk/parse/dependencygraph.py]: Fixed

**Comment by stevenbird** [comment]
@rmalouf – this might be the first instance of a parser which can return phrase-structure trees or dependency trees, which means I'd like to be sure that ad hoc polymorphism is the best way to go (instead of, say, providing `parse` and `depparse` methods or some such solution). Do you want to argue for the polymorphism approach? @heatherleaf do you have an opinion, in light of your earlier input to the parser API in #620?

**Comment by rmalouf** [comment]
No, I don't.  The version of this that I've been using actually had two classes, StanfordParser and StanfordDependencyParser.  I combined them just to minimize the changes, but I'm happy to split them again.  Any preferences?

**Comment by stevenbird** [comment]
I like that solution. So two classes both inheriting `ParserI`, but where `StanfordParser.parse` vs `StanfordDependencyParser.parse` return `Tree` vs `DependencyGraph`.

File: nltk/parse/stanford.py [hunk 2]
```diff
@@ -237,6 +188,114 @@ def _execute(self, cmd, input_, verbose=False):
...
+    ... ))],[]) # doctest: +NORMALIZE_WHITESPACE
+    [[((u'jumped', u'VBD'), u'nsubj', (u'fox', u'NN')), ((u'fox', u'NN'), u'det', (u'The', u'DT')),
+    ((u'fox', u'NN'), u'amod', (u'quick', u'JJ')), ((u'fox', u'NN'), u'amod', (u'brown', u'JJ')),
+    ((u'jumped', u'VBD'), u'nmod', (u'dog', u'NN')), ((u'dog', u'NN'), u'case', (u'over', u'IN')),
+    ((u'dog', u'NN'), u'det', (u'the', u'DT')), ((u'dog', u'NN'), u'amod', (u'lazy', u'JJ'))]]
+
+    """
+
+    _OUTPUT_FORMAT = 'conll2007'
+    _TO_TREE = DependencyGraph
```
Comment (dimazest) [inline nltk/parse/stanford.py]: this assignment look strange, there might be problems due to bound methods. Maybe defining a method is a nicer way of doing this.

File: nltk/parse/dependencygraph.py [hunk 3]
```diff
@@ -334,6 +334,9 @@ def extract_10_cells(cells):
             # Make sure that he fake root node has labeled dependencies.
             if (cell_number == 3) and (head == 0):
                 rel = 'ROOT'
+            # Stanford parser doesn't capitalize ROOT
```
Comment (dimazest) [inline nltk/parse/dependencygraph.py]: Actually, #1049 might be a cleaner solution.

File: nltk/parse/stanford.py (hunk 2 above)
Comment (rmalouf) [inline nltk/parse/stanford.py]: Strange how?

File: nltk/parse/dependencygraph.py (hunk 3 above)
Comment (rmalouf) [inline nltk/parse/dependencygraph.py]: That would be better! But, it looks like only malt.py got changed and the necessary changes to dependencygraph.py didn't make it in to #1049.

Comment (dimazest) [inline nltk/parse/dependencygraph.py]: That's strange, I've created #1131 to restore that code.

**Comment by stevenbird** [comment]
Thanks @rmalouf

**Pull request merged**